doc/tutorial/single_cell_bluepyopt.rst

```diff
@@ -0,0 +1,288 @@
+   axon_root = morpho.segment_tree.tag_roots(2)[0]
+
+   # prune morphology at axon root
+   pruned_st, axon_st = morpho.segment_tree.split_at(axon_root)
+
+Then we can determine the parent of the axon root to join the replacement onto, which we load from ACC. Alternatively, the user can build/supply a custom replacement here.
+
+.. code-block:: python
+
+   # parent segment to join onto
+   axon_parent = morpho.segment_tree.parents[axon_root]
+
+   # load replacement morphology
+   axon_replacement = arbor.load_component(
+      cell_json['morphology']['replace_axon']).component
+   axon_replacement_st = replacement.to_segment_tree()
+
+   # pruned segment tree with axon replacement
+   joined_st = pruned_st.join_at(
+      axon_parent, axon_replacement_st)
+
+   # final morphology
+   morpho_axon_replaced = arbor.morphology(joined_st)
+
+The steps presented here are included when loading an exported model with the above ``read_acc`` function from BluePyOpt.
+
+
+Define locations in the GUI
+---------------------------
+
+BluePyOpt makes use of the Neuron simulator's features to define custom locations for stimuli and voltage probes that can be challenging to translate to Arbor. As an example, the layer-5 pyramidal cell optimization problem has multiple voltage probes located distally on the apical dendrite at a fixed distance from the soma as described in `config/protocols.json <https://github.com/BlueBrain/BluePyOpt/blob/master/examples/l5pc/config/protocols.json>`_.
+
+To describe these locations accurately in Arbor, we will use the Arbor GUI. For this purpose, we create an :class:`arbor.cable_cell` and export it to ACC so that it can be loaded in the GUI,
+
+.. code-block:: python
+
+   cell = arbor.cable_cell(morpho, labels, decor)
+   arbor.write_component(cell, 'path/to/l5pc_cable_cell.acc')
+
+After loading this in file in the GUI with ``File > Cable cell > Load``, we can click the ``Locations`` and ``Cell`` tab to obtain the following overview.
+
+.. figure:: single_cell_bluepyopt_gui_l5pc_overview.png
+    :width: 800
+    :align: center
+
+As a first attempt to formulate the location at 660 µm from the soma, we can use
+
+.. code-block::
+
+   (distal-translate (on-components 0.5 (region "soma")) 660)
+
+and since we're only interested in the apical dendrite, we can wrap this in ``(restrict ... (region "apic"))``.
+
+Unfortunately, for this particular morphology, this does not result in a concrete location as can be seen in the Arbor GUI below (Locset A, violet, is empty). Alternatively, the method :func:`arbor.cable_cell.locations` will also return an empty list of concrete locations for this expression. To obtain a valid location instead, we can replace ``(on-components 0.5 (region "soma"))`` by the proximal boundary of the apical dendrite with ``(proximal (region "apic"))`` (Locset B, orange). The restrict operation to the apical region is no longer necessary. But now, the resulting concrete location is non-unique.
```

Comment:
An explanation for why no concrete location is found is missing. Is there one?

Reply:
And why is none-uniqueness an issue? Are locations in ``config/protocols.json`` supposed to be unique?